ent/encryption/encrypted_file_impl.cc

```diff
@@ -518,6 +523,9 @@ class indirect_encrypted_file_impl : public file_impl {
            });
         });
     }
+    future<struct stat> statat(std::string_view name, int flags = 0) override {
+        throw std::runtime_error("statat is not supported on encrypted file");
+    }
```

Comment:
Why do we have code development on a submodule update?

Reply:
Generic file_impl only remembers alignments. Maybe something like that would work:

```c++
future<stat> file_impl::statat(std::string_view name, unsigned flags) {
    auto nfd = this->fcntl(F_DUPFD);
    struct stat st;
    ::fstatat(nfd, name);
    ::close(nfd);
    co_return st;
}
```

Reply:
There's no fd for file_impl, it's not tied to posix files.

Reply:
> If a directory remembers its own name, then statat() can concatenate the directory path with the input path and call stat(). So we can provide a default implementation.

The stat method does not depend on a name. posix_file_impl::stat just uses fstat using the file descriptor.
For directories, if we had the full path (which we'll need to generate if the name we're given is relative), statat could use the high level `file_stat` as you said, but I don't see why we need to do that given posix_file_impl already implements statat efficiently using the fd based syscall.